src/rpp/jni_cpp.h

```diff
@@ -96,8 +103,9 @@ namespace rpp { namespace jni {
             {
                 // reconstruct
                 auto* env = getEnv();
-                if (obj) env->DeleteLocalRef(obj);
-                obj = env->NewLocalRef(r.obj);
+                reset();
+                isGlobal = r.isGlobal;
+                obj = isGlobal ? env->NewGlobalRef(r.obj) : env->NewLocalRef(r.obj);

```

Comment:
`r.obj` could be null.
I think this could be moved to some utility function since it is reused in copy constructor as well